src/workers/forum_automod.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime, timezone, timedelta
+import discord
+import logging
+import asyncio
+from settings import get_settings
+from util import log_info_and_embed, send_dm
+from .helper import create_automod_embed
+from discord.ext import tasks
+from discord.utils import snowflake_time
+
+settings = get_settings()
+logger = logging.getLogger(__name__)
+
+
+@tasks.loop(hours=24)
+async def autodelete_threads(self):
+    guild = self.get_guild(settings.guild_id)
+    if guild is None:
+        return
+
+    for channel_id, duration in settings.automod_inactivity.items():
+        channel = guild.get_channel(channel_id)
+        if channel is None:
+            continue
+
+        for thread in channel.threads:
+            last_post = thread.last_message_id
+
+            if thread.flags.pinned:
+                continue
```

Comment:
let's move this to the very top of the for loop, group `last_post` to be right above the comment of `starter_message`, and add a comment inside saying "skip the for loop if the thread is pinned"

Reply:
er sorry i meant the comment being inside the if block right before continue